server/isomorphic-routing/index.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Internal dependencies
+ */
+import i18n from 'lib/mixins/i18n';
+import { serverRender } from 'render';
+import { createReduxStore } from 'state';
+
+export function serverRouter( expressApp, mapExpressToPageContext ) {
+	return function( route, ...middlewares ) {
+		expressApp.get( route, combinedMiddlewares( mapExpressToPageContext, middlewares ) );
+	}
+}
+
+function combinedMiddlewares( mapExpressToPageContext, middlewares ) {
+	return function( req, res ) {
+		let context = mapExpressToPageContext( req );
+		context = getEnhancedContext( context, req, res );
+		applyMiddlewares( context, ...[
+			setUpRoute,
+			...middlewares,
+			serverRender
+		] );
+	}
+}
+
+function getEnhancedContext( context, req, res ) {
+	return Object.assign( {}, context, {
+		isLoggedIn: req.cookies.wordpress_logged_in,
+		isServerSide: true,
+		path: req.path,
+		params: Object.assign( {}, context.params, req.params ),
+		query: {},
+		store: createReduxStore(),
+		res,
+		url: req.url,
+	} );
+}
+
+function applyMiddlewares( context, ...middlewares ) {
+	const liftedmiddlewares = middlewares.map( middleware => next => middleware( context, next ) );
+	compose( ...liftedmiddlewares )();
+}
+
+function compose( ...functions ) {
```

Comment:
(For the record, `flowRight` wasn't fit for a drop-in replacement. It was agreed that having this local implementation is fine.)